common-theurgy.lic

```diff
@@ -0,0 +1,226 @@
+    'completed this commune too recently'
+  ] unless defined?(COMMUNE_ERRORS)
+
+  DEVOTION_LEVELS = [
+    'You sense nothing special from your communing',
+    'You feel unclean and unworthy',
+    'You close your eyes and start to concentrate',
+    'You call out to your god, but there is no answer',
+    'After a moment, you sense that your god is barely aware of you',
+    'After a moment, you sense that your efforts have not gone unnoticed',
+    'After a moment, you sense a distinct link between you and your god',
+    'After a moment, you sense that your god is aware of your devotion',
+    'After a moment, you sense that your god knows your name',
+    'After a moment, you sense that your god is pleased with your devotion',
+    'After a moment, you see a vision of your god, though the visage is cloudy',
+    'After a moment, you sense a slight pressure on your shoulder',
+    'After a moment, you see a silent vision of your god',
+    'After a moment, you see a vision of your god who calls to you by name, "Come here, my child',
+    'After a moment, you see a vision of your god who calls to you by name, "My child, though you may',
+    'After a moment, you see a crystal-clear vision of your god who speaks slowly and deliberately',
+    'After a moment, you feel a clear presence like a warm blanket covering you'
+  ] unless defined?(DEVOTION_LEVELS)
+
+  def has_holy_water?(theurgy_supply_container, water_holder)
+    DRCI.get_item?(water_holder, theurgy_supply_container)
+    DRCI.inside?('holy water', water_holder)
+    DRCI.put_away_item?(water_holder, theurgy_supply_container)
+  end
+
+  def has_holy_oil?(theurgy_supply_container)
+    DRCI.have_item_by_look?('holy oil', theurgy_supply_container)
+  end
+
+  def has_incense?(theurgy_supply_container)
+    DRCI.have_item_by_look?('incense', theurgy_supply_container)
+  end
+
+  def has_jalbreth_balm?(theurgy_supply_container)
+    DRCI.have_item_by_look?('jalbreth balm', theurgy_supply_container)
+  end
+
+  def buying_cleric_item_requires_bless?(town, item_name)
+    town_theurgy_data = get_data('theurgy')[town]
+    return if town_theurgy_data.nil?
+
+    item_shop_data = town_theurgy_data["#{item_name}_shop"]
+    return if item_shop_data.nil?
+
+    return item_shop_data['needs_bless']
+  end
+
+  def buy_cleric_item?(town, item_name, stackable, num_to_buy, theurgy_supply_container)
+    town_theurgy_data = get_data('theurgy')[town]
+    return false if town_theurgy_data.nil?
+
+    item_shop_data = town_theurgy_data["#{item_name}_shop"]
+    return false if item_shop_data.nil?
+
+    DRCT.walk_to(item_shop_data['id'])
+    if stackable
+      num_to_buy.times do
+        buy_single_supply(item_name, item_shop_data)
+        if DRCI.get_item?(item_name, theurgy_supply_container)
+          DRC.bput("combine #{item_name} with #{item_name}",
+               'You combine', 'You can\'t combine', 'You must be holding')
```

Comment:
Not sure if the newline was on purpose?
```suggestion
          DRC.bput("combine #{item_name} with #{item_name}", 'You combine', 'You can\'t combine', 'You must be holding')
```

Reply:
Eh, it just splits up the matches from the command for readability on the long line, but it doesn't matter.